NSTEPS/solution.ex

```diff
@@ -0,0 +1,30 @@
+defmodule ConversionUtils do
+  def convert_to(s, :integer) do
+    Integer.parse(s)
+    |> elem(0)
+  end
+end
+
+defmodule IOUtils do
+  def parse_input_line(l) do
+    String.trim(l)
+    |> String.split(" ")
+    |> Enum.map(&ConversionUtils.convert_to(&1, :integer))
+  end
+end
+
+defmodule StepCounter do
+  defp _steps(x, x) when x >= 0, do: 2 * x - rem(x, 2)
+  defp _steps(x, y) when x >= 2 and y == x - 2, do: 2 * (x - 1) - rem(x, 2)
```

Comment:
What do you think about to abstract the function `- rem(x, 2)` and apply to every `_steps` ?

Reply:
Updated